backend/spiritual_avatar_generation_engine.py

```
@@ -78,19 +89,28 @@ async def _generate_audio(self, text: str, voice_id: str) -> str:
+                bucket_name=bucket_name,
+                file_path_in_bucket=file_name_in_bucket,
+                file=audio_content,
+                content_type="audio/mpeg"
+            )
+            
+            logger.info(f"🎤 Successfully generated and uploaded audio to: {public_url}")
+            return public_url
 
         except httpx.HTTPStatusError as e:
             logger.error(f"ElevenLabs API error: {e.response.status_code} - {e.response.text}", exc_info=True)
             raise HTTPException(status_code=500, detail="Failed to generate audio for avatar.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)